backend/models/Vehicle.js

```diff
@@ -35,4 +38,11 @@ const VehicleSchema = new mongoose.Schema({
   properties: Object,
 })
 
+// cascade delete comments, history, vehiclesale
+VehicleSchema.pre('remove', async function(next) {
+  await Comment.deleteMany({ vehicle: this._id });
+  await History.deleteMany({ vehicle: this._id });
+  await VehicleSale.deleteMany({ vehicle: this._id });
```

Comment:
Looks good but I think we're missing to also delete all the events that are associated with the vehicles.

Reply:
Right that's a good point, will add this

Reply:
Added in this [commit](https://github.com/purplesmurf1998/Autotracks/pull/86/commits/cafd5ae8a3a1782cea668e1bd3f3b2a898b8ffcc)